bokeh/io.py

```diff
@@ -57,10 +57,29 @@
 #-----------------------------------------------------------------------------
 
 class _CommsHandle(object):
+
+    _json = {}
+
     def __init__(self, comms, doc, json):
+        self._cellno = None
+        try:
+            from IPython import get_ipython
+            ip = get_ipython()
+            hm = ip.history_manager
+            p_prompt = list(hm.get_tail(1, include_latest=True))[0][1]
+            self._cellno = p_prompt
+        except:
```

Comment:
In understand that and `except Exception` is the way to go. Normally I would suggest being a lot more specific, but in this case there is no luck for that.